rfcs/002-extensible-routing.md

```diff
@@ -0,0 +1,187 @@
+- Start Date: 2019/06/02
+- RFC PR: (leave this empty)
+- Tracking Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+This RFC tries to determine if extensible routing is needed in Tide, and if so
+what approach should be used.
+
+Note this is a continuation of [#271](https://github.com/rustasync/tide/issues/271).
+
+# Motivation
+[motivation]: #motivation
+
+Discussions around PRs [#254](https://github.com/rustasync/tide/pull/254) (largely
+on discord) and [#258](https://github.com/rustasync/tide/pull/258) addressed the question
+of being able to replace or change the implementation of routing within Tide to
+address different needs for different applications. The existing routing within
+Tide has issues that need addressing ([#12](https://github.com/rustasync/tide/issues/12)) and how these issues are resolved tie
+into the question of extensible routing.
```

Comment:
I'd like to see more in terms of motivation here. Could you repeat what the problems are stated in the links?

Reply:
Also the stakeholders section talks about "users with specialized routing requirements". It'd be good to lay out here what those requirements are.